src/handlers/gestures/GestureDetector.tsx

```diff
@@ -79,20 +79,42 @@ function dropHandlers(preparedGesture: GestureConfigReference) {
   }
 }
 
+function checkGestureCallbacksForWorklets(gesture: GestureType) {
+  // if a gesture is explicitly marked to run on the JS thread there is no need to check
+  // if callbacks are worklets as the user is aware they will be ran on the JS thread
+  if (gesture.config.runOnJS) {
+    return;
+  }
+
+  const areAllWorklets = gesture.handlers.isWorklet.reduce(
+    (prev, current) => prev && (current === undefined || current === true),
+    true
+  );
+
+  const areAllNotWorklets = gesture.handlers.isWorklet.reduce(
+    (prev, current) => prev && (current === undefined || current === false),
+    true
+  );
+
+  if (!areAllWorklets && !areAllNotWorklets) {
+    console.error(
+      `[RNGestureHandler] Some of the callbacks in a gesture are worklets and some are not. Either make sure that all calbacks are marked ad 'worklet' if you wish to run them on the UI thread or use 'runOnJS(true)' modifier on a gesture explicitly to run all callbacks on the JS thread`
```

Comment:
```suggestion
      `[RNGestureHandler] Some of the callbacks in a gesture are worklets and some are not. Either make sure that all calbacks are marked as 'worklet' if you wish to run them on the UI thread or use 'runOnJS(true)' modifier on a gesture explicitly to run all callbacks on the JS thread.`
```

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/1862/commits/4ea5e4db106f1e93a8de86ebe1d2ebf0946e5097.